src/NHibernate/Cfg/Configuration.cs

```diff
@@ -1299,33 +1291,22 @@ protected virtual void ConfigureProxyFactoryFactory()
 		/// <returns>An <see cref="ISessionFactory" /> instance.</returns>
 		public ISessionFactory BuildSessionFactory()
 		{
-			var dynamicDialectMapping = mapping;
 			// Use a mapping which does store the dialect instead of instantiating a new one
 			// at each access. The dialect does not change while building a session factory.
 			// It furthermore allows some hack on NHibernate.Spatial side to go on working,
 			// See nhibernate/NHibernate.Spatial#104
-			mapping = new StaticDialectMappingWrapper(mapping);
-			try
-			{
-				ConfigureProxyFactoryFactory();
-				SecondPassCompile();
-				Validate();
-				Environment.VerifyProperties(properties);
-				Settings settings = BuildSettings();
+			var m = BuildMapping();
+			var settings = BuildSettings();
+			var mapping = new StaticDialectMappingWrapper(m, settings.Dialect);
```

Comment:
`SessionFactoryImpl` will use dialect from session, so use it here as well.

Reply:
Considering that this static wrapping is still required I don't think field removal is actually necessary.

Reply:
Wrap in static wrapper is not required. It is possible to remove it entirely. 

The mapping field is not need as well as it itself is just a wrapper around configuration field.

Reply:
I've decided to remove mapping field to prevent further misuses of the same nature. I've initially started with exact code you have in your PR.

Reply:
Yeah I got it. I'm just saying that with required wrapping it's not any better. You can still create mapping and forgot to wrap it. Runtime exception is not very nice way to report about it.